hledger/test/balance/budget.test

```diff
@@ -50,7 +50,6 @@ Budget performance in 2016-12-01..2016-12-03:
                   ||          2016-12-01          2016-12-02           2016-12-03 
 ==================++==============================================================
  assets:cash      || $-10 [ 40% of $-25]  $-14 [56% of $-25]  $-51 [204% of $-25] 
- expenses         ||  $10 [ 40% of  $25]   $14 [56% of  $25]   $51 [204% of  $25] 
```

Comment:
Another behaviour change, worth noting with a ! in message.

The parent "expenses" account is not shown, because there's no explicit budget goal for it, and because we're in list mode ? So if we want to see aggregated budget performance, tree mode will be needed. Ok I guess.

Why is it still shown in the previous test ?

Reply:
Let me look into this.

Reply:
It looks like the reason is as follows:

Unless called with `-E`, then a budget report will count any unbudgeted subaccounts against their earliest budgeted parent. So both `expenses:cab` and `expenses:movies` are rolled up to `expenses`. Even though `expenses` doesn't have a budget itself, it gets the sum of the budgets of its subaccounts.

I'm not sure how I feel about this behaviour, but I think it's out of scope for this PR.